index.php

```diff
@@ -56,16 +106,76 @@ function gutenberg_register_scripts() {
  * @since 0.1.0
  */
 function gutenberg_scripts_and_styles( $hook ) {
-	if ( 'toplevel_page_gutenberg' === $hook ) {
-		// Scripts
-		wp_register_script( 'gutenberg-content', plugins_url( 'post-content.js', __FILE__ ) );
-		wp_enqueue_script( 'wp-editor', plugins_url( 'editor/build/index.js', __FILE__ ), array( 'wp-blocks', 'wp-element', 'gutenberg-content' ), false, true );
-		wp_add_inline_script( 'wp-editor', 'wp.editor.createEditorInstance( \'editor\', { content: window.content } );' );
-
-		// Styles
-		wp_enqueue_style( 'wp-editor-font', 'https://fonts.googleapis.com/css?family=Noto+Serif:400,400i,700,700i', false );
-		wp_enqueue_style( 'wp-editor', plugins_url( 'editor/build/style.css', __FILE__ ) );
+	if ( 'toplevel_page_gutenberg' !== $hook ) {
+		return;
 	}
+
+	/**
+	 * Scripts
+	 */
+
+	// The editor code itself
+	wp_enqueue_script(
+		'wp-editor',
+		plugins_url( 'editor/build/index.js', __FILE__ ),
+		array( 'wp-blocks', 'wp-element', 'gutenberg-content' ),
+		false, // $ver
+		true   // $in_footer
+	);
+
+	// Temporary test content
+	wp_register_script(
+		'gutenberg-content',
+		plugins_url( 'post-content.js', __FILE__ )
+	);
+
+	// Load an actual post if an ID is specified
+	$post_to_edit = null;
+	if ( isset( $_GET['post_id'] ) && (int) $_GET['post_id'] > 0 ) {
+		$request = new WP_REST_Request(
+			'GET',
+			sprintf( '/wp/v2/posts/%d', (int) $_GET['post_id'] )
+		);
+		$request->set_param( 'context', 'edit' );
+		$response = rest_do_request( $request );
+		if ( 200 === $response->get_status() ) {
+			$post_to_edit = $response->get_data();
+		}
+	}
+
+	// Initialize the post data...
+	if ( $post_to_edit ) {
+		// ...with a real post
+		wp_add_inline_script(
+			'wp-editor',
+			'wp.editor.post = ' . json_encode( $post_to_edit )
```

Comment:
Why might we want to construct multiple instances of the editor?

Do we want to do that refactoring to a single editor instance now?  I'm guessing probably not, constructing an editor without depending on global state seems like a more desirable code structure regardless of how we end up using it.

If we do use a global variable we can use `wp_localize_script` which seems slightly more _preferred_ for injecting script data, but doesn't support object paths with `.` in them.

I don't have a strong preference here, but if we're going to change something on this PR, here's what I would do:

- Rename the current post to `_wpGutenbergPost` as you suggested
- Also rename the "default" test post content from `window.content` to something like `_wpGutenbergTestPost` (or, be slightly more clever and set this to `_wpGutenbergPost` if no post is loaded, simplifying the code path for initializing the editor).